session/channel.go

```diff
@@ -250,18 +254,21 @@ func (ch *Channel) SubChUpdates(notifier perun.ChUpdateNotifier) error {
 }
 
 // UnsubChUpdates implements chAPI.UnsubChUpdates.
-func (ch *Channel) UnsubChUpdates() error {
+func (ch *Channel) UnsubChUpdates() perun.APIErrorV2 {
 	ch.Debug("Received request: channel.UnsubChUpdates")
 	ch.Lock()
 	defer ch.Unlock()
 
 	if ch.status == closed {
-		return perun.ErrChClosed
+		apiErr := perun.NewAPIErrV2FailedPreCondition(ErrChClosed.Error())
+		ch.WithFields(perun.APIErrV2AsMap("UnsubChUpdates", apiErr)).Error(apiErr.Message())
+		return apiErr
 	}
 
 	if ch.chUpdateNotifier == nil {
-		ch.Error(perun.ErrNoActiveSub)
-		return perun.ErrNoActiveSub
+		apiErr := perun.NewAPIErrV2ResourceNotFound("unsubscription to channel updates", ch.ID(), ErrNoActiveSub.Error())
```

Comment:
```suggestion
		apiErr := perun.NewAPIErrV2ResourceNotFound("subscription to channel updates", ch.ID(), ErrNoActiveSub.Error())
```

It is `subscription to channel updates` that is not found.